src/arrays/constructors.jl

```diff
@@ -16,4 +16,16 @@
     one_at_ind!(arr, i) = setindex!(arr, one(eltype(arr)), i)
     single_coeff(i, lens...) = one_at_ind!(zeros(lens), i)
 
-export statevec
\ No newline at end of file
+function coherentstatevec(n::Int,alpha::Number;method="direct")
+    s = rand(n)
+    z = zeros(QuArray(s))
+    for i in [1:n]
+        z = z + ((alpha^(i-1))/sqrt(factorial(i)))*statevec(i, FiniteBasis(n))
```

Comment:
... And it should be `factorial(i-1)`. Note that creating the coefficients like that is probably not a good idea if n is large. If one would do this it might be better (I.e. More efficient) to use a recursive prescription. Something like (untestet!)
```
cs = zeros(n)
cs[1]=one(alpha)
For i in 2:n
   cs[i] = alpha/sqrt(i-1)*cs[i-1]
end
QuArray(cs)
```
Also note that the resulting state is not normalized (only approximately), because the formula for the elements in Fock space assumes infinite n. But I would anyways go with the other definition below and only use the formula in the test.